apps/api/src/utils/aliases.ts

```diff
@@ -0,0 +1,142 @@
+import { type ApplicationContentsResponse, type ApplicationResponseData } from '@pcgl-daco/data-model/src/types.js';
+import { type UpdateEditApplicationRequest } from '@pcgl-daco/validation';
+import lodash from 'lodash';
+
+/** Used in filter functions for alias utilities below
+ * @param key Current key to validate
+ * @param omittedKeys List of keys to remove from output
+ */
+const filterOmittedKeys = (key: string, omittedKeys: string[]) => !(omittedKeys.includes(key) || key === 'id');
+
+/**
+ * Helper function to convert Postgres snake_case to FE camelCase
+ * Generics allow usage w/ multiple input/output combinations
+ * @param data Original Database record with snake case keys to convert
+ * @param omittedKeys List of keys to remove from output for cases where a partial record is returned
+ * @returns ResponseRecord - Generic type representing new record with updated camelCase keys
+ */
+export const aliasToResponseData = <
+	DatabaseRecord extends Record<string, any>,
+	ResponseRecord extends Record<string, any>,
+>(
+	data: DatabaseRecord,
+	omittedKeys: string[] = [],
+): ResponseRecord => {
+	type snakeCaseKey = string & keyof DatabaseRecord;
+	const allKeys = Object.keys(data).map((key): snakeCaseKey => key);
+
+	const filteredKeys = allKeys.filter((key) => {
+		return filterOmittedKeys(String(key), omittedKeys);
+	});
+
+	const responseData = filteredKeys.reduce((acc, key) => {
+		const aliasedKey = lodash.camelCase(key);
+		const value = data[key];
+		const accumulator = { ...acc, [aliasedKey]: value };
+		return accumulator;
+	}, {}) as ResponseRecord;
+
+	return responseData;
+};
+
+/**
+ * Helper function to convert FE camelCase to Postgres snake_case
+ * Generics allow usage w/ multiple input/output combinations
+ * @param data Original Request record with camelCase keys to convert
+ * @param omittedKeys List of keys to remove from output for cases where a partial record is returned
+ * @returns DatabaseRecord - Generic type representing new record with updated snake_case keys
+ */
+export const aliasToDatabaseData = <
+	RequestRecord extends Record<string, any>,
+	DatabaseRecord extends Record<string, any>,
+>(
+	data: RequestRecord,
+	omittedKeys: string[] = [],
+): DatabaseRecord => {
+	type camelCaseKey = string & keyof RequestRecord;
+	const allKeys = Object.keys(data).map((key): camelCaseKey => key);
+
+	const filteredKeys = allKeys.filter((key) => {
+		return filterOmittedKeys(String(key), omittedKeys);
+	});
+
+	const databaseData = filteredKeys.reduce((acc, key) => {
+		const aliasedKey = lodash.snakeCase(key);
+		const value = data[key];
+		const accumulator = { ...acc, [aliasedKey]: value };
+		return accumulator;
+	}, {}) as DatabaseRecord;
+
+	return databaseData;
+};
+
+/** Convenience function for specific alias utils input/output scenarios */
+export const aliasApplicationRecord = (data: JoinedApplicationRecord): ApplicationResponseData => {
```

Comment:
This is an example where you're going from a complete database record to a partial UI record and so some of the keys are omitted